src/EncompassRest/ResourceLocks/ResourceLocks.cs

```diff
@@ -0,0 +1,132 @@
+            queryParameters.Add(new QueryParameter("resourceType", resourceType));
+            queryParameters.Add(new QueryParameter("resourceId", resourceId));
+
+            return await GetAsync<ResourceLock>(lockId, queryParameters.ToString(), nameof(GetResourceLockAsync), lockId, cancellationToken);
+        }
+
+        public Task<string> GetResourceLockRawAsync(string lockId, string queryString, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(queryString, nameof(queryString));
+            Preconditions.NotNullOrEmpty(lockId, nameof(lockId));
+
+            return GetRawAsync(lockId, queryString, nameof(GetResourceLockRawAsync), lockId, cancellationToken);
+        }
+
+        public Task<List<ResourceLock>> GetResourceLocksAsync(string resourceId, EntityType resourceType, CancellationToken cancellationToken = default) => GetResourceLocksAsync(resourceId, resourceType.ToString(), cancellationToken);
+        public async Task<List<ResourceLock>> GetResourceLocksAsync(string resourceId, string resourceType,  CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resourceType, nameof(resourceType));
+
+            var queryParameters = new QueryParameters();
+            queryParameters.Add(new QueryParameter("resourceType", resourceType));
+            queryParameters.Add(new QueryParameter("resourceId", resourceId));
+
+            return await GetAsync<List<ResourceLock>>(null, queryParameters.ToString(), nameof(GetResourceLocksAsync), null, cancellationToken);
+        }
+
+        public Task<string> GetResourceLocksRawAsync(string queryString, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(queryString, nameof(queryString));
+
+            return GetRawAsync(null, queryString, nameof(GetResourceLockRawAsync), null, cancellationToken);
+        }
+
+        public Task<string> LockResourceAsync(ResourceLock resourceLock, CancellationToken cancellationToken = default) => LockResourceAsync(resourceLock, false, false, cancellationToken);
+        public Task<string> LockResourceAsync(ResourceLock resourceLock, bool force, CancellationToken cancellationToken = default) => LockResourceAsync(resourceLock, force, false, cancellationToken);
+        public Task<string> LockResourceAsync(ResourceLock resourceLock, bool force, bool populate, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNull(resourceLock, nameof(resourceLock));
+
+            var queryParameters = new QueryParameters();
+            if (force) queryParameters.Add(new QueryParameter("force", "true"));
+            if (populate) queryParameters.Add(new QueryParameter("view", "entity"));
+
+            return PostPopulateDirtyAsync<ResourceLock>(null, (force || populate) ? queryParameters.ToString() : null, resourceLock, nameof(LockResourceAsync),populate, cancellationToken);
```

Comment:
The `force || populate` check is unneeded. If there are no parameters it will just return `string.Empty`.

Also add space between `nameof(LockResourceAsync),` and `populate`